src/main/java/at/hannibal2/skyhanni/features/misc/CurrentPing.kt

```diff
@@ -0,0 +1,46 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.api.event.HandleEvent
+import at.hannibal2.skyhanni.api.hypixelapi.HypixelEventApi
+import at.hannibal2.skyhanni.events.SecondPassedEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import net.hypixel.modapi.packet.impl.clientbound.ClientboundPingPacket
+import net.hypixel.modapi.packet.impl.serverbound.ServerboundPingPacket
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+
+@SkyHanniModule
+object CurrentPing {
+
+    private var lastPingRequested = SimpleTimeMark.farPast()
+    private val previousPings = mutableListOf<Duration>()
+    private var waitingForPacket = false
+    var averagePing = Duration.ZERO
+
+    fun onPingPacket(packet: ClientboundPingPacket) {
+        waitingForPacket = false
+
+        if (previousPings.size > 5) {
+            previousPings.dropLast(1)
+        }
+        previousPings.add(SimpleTimeMark.now() - lastPingRequested)
+        averagePing = (previousPings.sumOf { it.inWholeMilliseconds } / previousPings.size).milliseconds
```

Comment:
maybe store the previous pings as ms in double directly, then we can use the existing `.average()` function instead

Reply:
also since we alr have ping/tps data, (maybe outside this pr), we could show instead of current values, the avg over last min/5min/15min like paper spigot /tps does